website/src/blogs/2018-11-27-experience-with-openebs-in-this-hacktoberfest.md

```diff
@@ -0,0 +1,57 @@
+
+## Help is available in the community if you need.
+
+When I decided to work on the project OpenEBS/Maya, I struggled to set up the project on my machine. I was using MacOS, and the build instructions were written for Linux OSes. People like [Satyam Zode](https://medium.com/@satyamz) and Prateek helped me to build the project by changing some lines in the `Makefile`.
+This project was written entirely in `Golang`. When I decided to work on this project, I didn’t even know how to write a hello world program in `Golang`. But people were there to help me whenever I got stuck.
+
+## Always consult with some maintainers before you add some external libraries to the project.
+
+I took the issue of adding bash completion to a CLI tool named mayactl, since Akshay has mentioned a library to do the same thing. I learned a bit of `Golang` from A Tour of Go. Then I tried to incorporate the bash completion library to mayactl. After a couple of failed attempt, I was able to implement the essential bash completion. In the review, [Ashutosh Kumar](https://medium.com/@sonasingh46) told me that if we add an external library to our project, we should also be able to maintain it. So, he asked me to decide on it.
+
+## You don’t need to know every bit to fix an issue.
+
+Instead of using some external library, people told me to look into the source code of kubectl to see how they implemented the bash completion. I dug into kubectl’s source and figured out the part where they performed bash completion. I copied it and made it compatible with mayactl.
+
+`Copying code is fine as long as you know what it does is.`
+
+You don’t even need to know a language to work on an issue. You can learn the language on the go. To fix the bash completion issue, all I need to know was three things:
+
+- A little bit of `Golang`
+- Know how bash completion works
+- Where to add the code inside mayactl
+
+I didn’t know all these things before I started. I learned this by doing it.
+I was happy by knowing that this PR was selected as the best PR of the week and I won an **Amazon Echo** from OpenEBS.
+
+[Tweet](https://twitter.com/proaksh/status/1056212237361115136/photo/1?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1056212237361115136&amp;ref_url=https%3A%2F%2Fblog.openebs.io%2Fmedia%2Fa4ac13e292477c21ec22a988cdcc3daf%3FpostId%3D64b9711a22f5)
+
+The next issue that I took was to add some features to a Common Integration Testing Framework (CITF) that OpenEBS is planning to use for all of their projects. I thought this is difficult to work on since I don’t know how OpenEBS works. But the people like Ashutosh, [Prince Rachit](https://medium.com/@princerachit) and [Akash Srivastava](https://medium.com/@srivastavaakash) explained the essential parts of OpenEBS that are required to work on the issue. Without completely understanding it, I worked on it. And with the help from the community, I could complete the tasks.
```

Comment:
```suggestion
The next issue that I took was to add some features to a Common Integration Testing Framework (CITF) that OpenEBS is planning to use for all of its projects. I thought this is difficult to work on since I don’t know how OpenEBS works. But the people like Ashutosh, [Prince Rachit](https://medium.com/@princerachit) and [Akash Srivastava](https://medium.com/@srivastavaakash) explained the essential parts of OpenEBS that are required to work on the issue. Without completely understanding it, I worked on it. And with the help from the community, I could complete the tasks.
```